modules/business-text/business-text.service.js

```diff
@@ -37,32 +39,122 @@ class BusinessTextService {
       };
     }
 
-    const text = await BusinessText.findByIdAndUpdate(id, businessText, {
+    const newPage = files.length
+      ? await this.pageImageReplaceHandler(businessText, files)
+      : businessText;
+    const oldPage = await this.getBusinessTextById(id);
+
+    if (oldPage) {
+      const newImages = this.imageFinder(newPage);
+      const oldImages = this.imageFinder(oldPage);
+
+      const imagesToDelete = oldImages.filter(
+        img => newImages.findIndex(newImg => newImg === img) === -1
+      );
+
+      if (imagesToDelete.length) {
+        await this.deleteNoNeededImages(imagesToDelete);
+      }
+    } else {
+      throw new Error(BUSINESS_TEXT_NOT_FOUND);
+    }
+
+    const text = await BusinessText.findByIdAndUpdate(id, newPage, {
       new: true,
     });
 
     return text || null;
   }
 
-  async addBusinessText(data) {
-    const pages = await this.checkBusinessTextExistByCode(data);
+  async addBusinessText(businessText, files) {
+    const existingPages = await this.checkBusinessTextExistByCode(businessText);
 
-    if (pages.length) {
+    if (existingPages.length) {
       throw new Error(BUSINESS_TEXT_ALREADY_EXIST);
     }
-    return new BusinessText(data).save();
+
+    let newPage = '';
+    if (files.length) {
+      newPage = await this.pageImageReplaceHandler(businessText, files);
+    }
+
+    return new BusinessText(newPage || businessText).save();
   }
 
   async deleteBusinessText(id) {
+    const oldPage = await this.getBusinessTextById(id);
+
+    if (oldPage) {
+      const imagesToDelete = this.imageFinder(oldPage);
+
+      if (imagesToDelete.length) {
+        await this.deleteNoNeededImages(imagesToDelete);
+      }
+    } else {
+      throw new Error(BUSINESS_TEXT_NOT_FOUND);
+    }
+
     const businessText = await BusinessText.findByIdAndDelete(id);
     if (businessText) {
       return businessText;
     }
-    throw new Error(BUSINESS_TEXT_NOT_FOUND);
   }
 
   async checkBusinessTextExistByCode(data) {
     return await BusinessText.find({ code: data.code });
   }
+
+  async pageImageReplaceHandler(page, files) {
+    const fileNames = files.map(({ file }) => file.filename);
+
+    const uploadResult = await uploadFiles(files);
+    const imagesResults = await Promise.allSettled(uploadResult);
+
+    const updatedPage = page;
```

Comment:
Why do we need this line?

Reply:
I agree, in this case - it's really a useless variable. So I changed it to `const updatedPage = {... page};`, because I would like to go further with a new name to understand